tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala

```diff
@@ -0,0 +1,296 @@
+  }
+
+  test("opaque-type-alias") {
+    runTestAssert[Stat]("opaque type F = X")(
+      Defn.OpaqueTypeAlias(
+        List(Mod.Opaque()),
+        pname("F"),
+        Nil,
+        Type.Bounds(None, None),
+        pname("X")
+      )
+    )(parseTempl)
+  }
+
+  test("opaque-type-bounded-alias") {
+    runTestAssert[Stat]("opaque type F <: A & B = AB")(
+      Defn.OpaqueTypeAlias(
+        List(Mod.Opaque()),
+        pname("F"),
+        Nil,
+        Type.Bounds(None, Some(Type.And(pname("A"), pname("B")))),
+        pname("AB")
+      )
+    )(parseTempl)
+  }
+
+  test("opaque-type-in-object") {
+    runTestAssert[Source]("object X { opaque type IArray[+T] = Array }")(
+      Source(
+        List(
+          Defn.Object(
+            Nil,
+            tname("X"),
+            tpl(
+              List(
+                Defn.OpaqueTypeAlias(
+                  List(Mod.Opaque()),
+                  pname("IArray"),
+                  List(
+                    Type.Param(
+                      List(Mod.Covariant()),
+                      Type.Name("T"),
+                      Nil,
+                      Type.Bounds(None, None),
+                      Nil,
+                      Nil
+                    )
+                  ),
+                  Type.Bounds(None, None),
+                  pname("Array")
+                )
+              )
+            )
+          )
+        )
+      )
+    )(parseSource)
+  }
+
+  test("trait-parameters") {
+    runTestAssert[Stat]("trait Foo(val foo: Int)(bar: Int)")(
+      Defn.Trait(
+        Nil,
+        pname("Foo"),
+        Nil,
+        Ctor.Primary(
+          Nil,
+          anon,
+          List(
+            List(tparamval("foo", "Int")),
+            List(tparam("bar", "Int"))
+          )
+        ),
+        tpl(Nil)
+      )
+    )
+  }
+
+  test("trait-parameters-generic") {
+    runTestAssert[Stat]("trait Foo[T](bar: T)")(
+      Defn.Trait(Nil, pname("Foo"), List(pparam("T")), ctorp(List(tparam("bar", "T"))), tpl(Nil))
+    )
+  }
+
+  test("class-parameters-using") {
+    runTestAssert[Stat]("trait A(using String)")(
+      Defn.Trait(Nil, pname("A"), Nil, ctorp(List(tparamUsing("", "String"))), tpl(Nil))
+    )
+
+    runTestAssert[Stat]("class A(using String)")(
+      Defn.Class(Nil, pname("A"), Nil, ctorp(List(tparamUsing("", "String"))), tpl(Nil))
+    )
+
+    runTestAssert[Stat]("case class A(a: Int)(using b: String)")(
+      Defn.Class(
+        List(Mod.Case()),
+        pname("A"),
+        Nil,
+        Ctor.Primary(Nil, anon, List(List(tparam("a", "Int")), List(tparamUsing("b", "String")))),
+        tpl(Nil)
+      )
+    )
+  }
+
+  test("inline-soft-keyword-pos") {
+    // as ident
+    runTestAssert[Stat]("def f(inline: String): Unit")(
+      Decl.Def(Nil, tname("f"), Nil, List(List(tparam("inline", "String"))), pname("Unit"))
+    )(parseTempl)
+
+    // as ident
+    runTestAssert[Stat]("inline def inline(inline: inline): inline")(
```

Comment:
:fearful:

Reply:
you could simply state `inline` is keyword and move on, but hey, soft keyword is much more fun 😒